src/components/editor-content/editor-content.tsx

```diff
@@ -159,184 +340,208 @@ const EditorContent: React.FC<Props> = () => {
+        </Typography>
+      </Box>
+      <Box className={ classes.datePickers }>
+        <DateRangePicker
+          scope={ scope }
+          dateFormat={ dateFormat }
+          selectedStartDate={ selectedStartDate }
+          selectedEndDate={ selectedEndDate }
+          startWeek={ startWeek }
+          endWeek={ endWeek }
+          startDateOnly={ startDateOnly }
+          datePickerView={ datePickerView }
+          onStartDateChange={ handleStartDateChange }
+          onEndDateChange={ handleEndDateChange }
+          onStartWeekChange={ handleStartWeekChange }
+          onEndWeekChange={ handleEndWeekChange }
+        />
+      </Box>
+    </>
   );
 
   /**
-   * Renders end year picker and week selector 
+   * Renders the filter component
    */
-  const renderEndYearPickerAndWeekSelector = () => (
-    <>
-      <FormControl variant="standard">
-        <MuiPickersUtilsProvider utils={ DateFnsUtils } >
-          <Grid className={ classes.timeFilterYearSelector }>
-            <KeyboardDatePicker
-              variant="inline"
-              views={[ FilterScopes.YEAR ]}
-              format="yyyy"
-              id="date-picker-year-end"
-              label={ strings.editorContent.selectYearEnd }
-              value={ selectedEndingDate }
-              onChange={ handleEndDateChange }
-              KeyboardButtonProps={{ "aria-label": `${strings.editorContent.filterStartingDate}` }}
-            />
-          </Grid>
-        </MuiPickersUtilsProvider>
-      </FormControl>
-      <FormControl variant="standard" className={ classes.selectWeekNumbers }>
-        <TextField
-          select
-          id="scope-select-outlined"
-          value={ endWeek }
-          onChange={ handleEndWeekChange }
-          label={ strings.editorContent.selectWeekEnd }
+  const renderFilter = () => {
+    if (!person) {
+      return (
+        <Paper
+          elevation={ 3 }
+          className={ classes.emptyFilterContainer }
         >
-          { renderWeekNumbers() }
-        </TextField>
-      </FormControl>
-    </>
-  );
+          <Typography style={{ fontStyle: "italic" }}>
+            { strings.editorContent.userNotSelected }
+          </Typography>
+        </Paper>
+      );
+    }
+
+    if (!personTotalTime || !displayedTotal || !displayedTimeData) {
+      return (
+        <Paper
+          elevation={ 3 }
+          className={ classes.emptyFilterContainer }
+        >
+          <Typography style={{ fontStyle: "italic" }}>
+            { strings.editorContent.noTimeEntries }
+          </Typography>
+        </Paper>
+      );
+    }
+
+    let timeRangeText = "";
+
+    switch (displayedTimeData.length) {
+      case 0:
+        timeRangeText = "";
+        break;
+      case 1:
+        timeRangeText = `${displayedTimeData[0].name}`;
+        break;
+      default:
+        timeRangeText = `(${displayedTimeData[0].name} - ${displayedTimeData[displayedTimeData.length - 1].name})`;
+        break;
+    }
```

Comment:
Move to own function and call it directly from render. Best place would probably be in TimeUtils file